lightning/src/ln/channel.rs

```diff
@@ -2689,12 +2731,13 @@ impl<Signer: Sign> Channel<Signer> {
 			}
 		}
 		for htlc in self.pending_outbound_htlcs.iter_mut() {
-			if let Some(fail_reason) = if let &mut OutboundHTLCState::RemoteRemoved(ref mut fail_reason) = &mut htlc.state {
-				Some(fail_reason.take())
-			} else { None } {
+			if let &mut OutboundHTLCState::RemoteRemoved(ref mut fail_reason) = &mut htlc.state {
 				log_trace!(logger, "Updating HTLC {} to AwaitingRemoteRevokeToRemove due to commitment_signed in channel {}.",
 					log_bytes!(htlc.payment_hash.0), log_bytes!(self.channel_id));
-				htlc.state = OutboundHTLCState::AwaitingRemoteRevokeToRemove(fail_reason);
+				// Grab the preimage, if it exists, instead of cloning
+				let mut reason = OutboundHTLCOutcome::Success(None);
+				mem::swap(fail_reason, &mut reason);
+				htlc.state = OutboundHTLCState::AwaitingRemoteRevokeToRemove(reason.into());
```

Comment:
nit: I believe the `.into()` here is unnecessary, and below.

Reply:
fixed